x/distribution/keeper/allocation.go

```diff
@@ -49,29 +51,41 @@ func (k Keeper) AllocateTokens(ctx context.Context, totalPreviousPower int64, bo
 	feeMultiplier := feesCollected.MulDecTruncate(voteMultiplier)
 
 	// allocate tokens proportionally to voting power
-	//
-	// TODO: Consider parallelizing later
-	//
-	// Ref: https://github.com/cosmos/cosmos-sdk/pull/3099#discussion_r246276376
+	var g errgroup.Group
+	// channel for remaining rewards
+	remainingCh := make(chan sdk.DecCoins, len(bondedVotes))
```

Comment:
I think this will be dependent on store v2 which allows parallel writes. Maybe v1 does too but im not 100% certain.